### DISCUSSION TITLE
Disable queue during deployment

### DISCUSSION THREAD

**Discussion by alexgit2k** [craftcms/cms#12683] [Ideas]
We are having issues during deployment because of the queue-worker which locks database-tables or is writing logfiles. Therefore I suggest that if [isSystemLive / CRAFT_IS_SYSTEM_LIVE](https://craftcms.com/docs/4.x/config/general.html#issystemlive) is set to false:
- Running queue-workers/-daemon will finish the current queue-job and then exit immediately
- New queue-workers/-daemon can not be started (or exit immediately)

This should be applied for [queue/run](https://craftcms.com/docs/4.x/console-commands.html#queue-run) (cronjob) as well as [queue/listen](https://craftcms.com/docs/4.x/console-commands.html#queue-listen) (daemon). So that after the deployment updates applied to the queue-handling itself are activated with the first start of the queue after the deployment.

**Comment by brandonkelly** [comment]
The system status is used for more than just deployments. We are planning on repurposing Craft’s “Maintenance Mode” in Craft 5 to be used for deployments (#9855), at which point it would make sense to pause queue execution as well.

**Reply by alexgit2k** [reply]
Then how about introducing a new variable like `CRAFT_IS_DEPLOYED` or dedicated variable for the queue like `CRAFT_QUEUE_IS_ACTIVE`?

There may still be pretty of time before Craft 5 is released ... last time I had 3 failed deployments in a row until I manually deactivated the queue-job.

**Reply by brandonkelly** [reply]
How exactly are you running your queue?

**Reply by alexgit2k** [reply]
As cron-job (accordingly to https://github.com/yiisoft/yii2-queue/blob/master/docs/guide/worker.md#cron):
`* * * * * cd /var/www/my_project/ && ./craft queue/run`
